pkgs/development/python-modules/dirsearch/default.nix

```diff
@@ -0,0 +1,95 @@
+{
+  lib,
+  fetchFromGitHub,
+  buildPythonPackage,
+  python,
+  # deps
+  /*
+    ntlm-auth is in the requirements.txt, however nixpkgs tells me
+    > ntlm-auth has been removed, because it relies on the md4 implementation provided by openssl. Use pyspnego instead.
+    Not sure if pyspnego is a drop in replacement.
+    The simple functionality dirsearch seems not to depend on this package.
+  */
+  #ntlm-auth,
+  #pyspnego,
+  beautifulsoup4,
+  certifi,
+  cffi,
+  chardet,
+  charset-normalizer,
+  colorama,
+  cryptography,
+  defusedxml,
+  idna,
+  jinja2,
+  markupsafe,
+  pyopenssl,
+  pyparsing,
+  pysocks,
+  requests,
+  requests-ntlm,
+  setuptools,
+  urllib3,
+}:
+
+buildPythonPackage rec {
+  pname = "dirsearch";
+  version = "0.4.3";
+
```

Comment:
Please set `pyproject = true` and `build-system = [ setuptools ]` to avoid legacy build.

> pyproject: Whether the pyproject format should be used. As all other formats are deprecated, you are recommended to set this to true. When you do so, pypaBuildHook will be used, and you can add the required build dependencies from build-system.requires to build-system. Note that the pyproject format falls back to using setuptools, so you can use pyproject = true even if the package only has a setup.py. When set to false, you can use the existing [hooks](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#setup-hooks) or provide your own logic to build the package. This can be useful for packages that don't support the pyproject format. When unset, the legacy setuptools hooks are used for backwards compatibility.

https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#buildpythonpackage-parameters-buildpythonpackage-parameters

Reply:
addressed in #365128